### DISCUSSION TITLE
Unmarshalling error invalid character '}'

### DISCUSSION THREAD

**Discussion by sidiney-crescencio** [argoproj-labs/argocd-notifications#329] [Q&A]
I sometimes face errors and therefore the notification is not sent, for now I'm only using microsoft teams as destination and  then after a few minutes it works just fine without any changes from my side.

I'm running argoprojlabs/argocd-notifications:v1.1.1

And I use these annotations for my apps 

      notifications.argoproj.io/subscribe.on-sync-succeeded.teams: channelName
      notifications.argoproj.io/subscribe.on-health-degraded.teams: channelName
      notifications.argoproj.io/subscribe.on-sync-failed.teams: channelName
      notifications.argoproj.io/subscribe.on-sync-status-unknown.teams: channelName

Any ideas on why this happens ? 

Thanks.

> 
>  time="2021-07-26T09:39:50Z" level=info msg="Trigger on-sync-succeeded result: [{[0].zxM90Et6k4Elb1-fHdjtDJq0xR0  [app-sync-succeeded] false}]" app=argocd/filebeat
> time="2021-07-26T09:39:50Z" level=info msg="Trigger on-health-degraded result: [{[0].zoG5zUjYGjimkwCGgIQGjx1hO5s  [app-health-degraded] false}]" app=argocd/filebeat
> time="2021-07-26T09:39:50Z" level=info msg="Trigger on-sync-failed result: [{[0].H9WjsqG1dKYm6njOZ7yUQYOA1Wk  [app-sync-failed] true}]" app=argocd/filebeat
> time="2021-07-26T09:39:50Z" level=info msg="Sending notification about condition 'on-sync-failed.[0].H9WjsqG1dKYm6njOZ7yUQYOA1Wk' to '{teams channelName}'" app=argocd/filebeat
> time="2021-07-26T09:39:50Z" level=error msg="Failed to notify recipient {teams channelName} defined in app argocd/filebeat: teams facts unmarshalling error invalid character '}' looking for beginning of object key string" app=argocd/filebeat
> time="2021-07-26T09:39:50Z" level=info msg="Trigger on-sync-status-unknown result: [{[0].6SzWb05EK-0v90hwjyytTbN7S6A  [app-sync-status-unknown] false}]" app=argocd/filebeat
> 
> time="2021-07-26T09:45:50Z" level=info msg="Start processing" app=argocd/filebeat
> time="2021-07-26T09:45:50Z" level=info msg="Trigger on-sync-failed result: [{[0].H9WjsqG1dKYm6njOZ7yUQYOA1Wk  [app-sync-failed] false}]" app=argocd/filebeat
> time="2021-07-26T09:45:50Z" level=info msg="Trigger on-sync-status-unknown result: [{[0].6SzWb05EK-0v90hwjyytTbN7S6A  [app-sync-status-unknown] false}]" app=argocd/filebeat
> time="2021-07-26T09:45:50Z" level=info msg="Trigger on-sync-succeeded result: [{[0].zxM90Et6k4Elb1-fHdjtDJq0xR0  [app-sync-succeeded] true}]" app=argocd/filebeat
> time="2021-07-26T09:45:50Z" level=info msg="Notification about condition 'on-sync-succeeded.[0].zxM90Et6k4Elb1-fHdjtDJq0xR0' already sent to '{teams channelName}'" app=argocd/filebeat
> time="2021-07-26T09:45:50Z" level=info msg="Trigger on-health-degraded result: [{[0].zoG5zUjYGjimkwCGgIQGjx1hO5s  [app-health-degraded] false}]" app=argocd/filebeat
> time="2021-07-26T09:45:50Z" level=info msg="Processing completed" app=argocd/filebeat

**Comment by ryota-sakamoto** [comment]
@sidiney-crescencio 
I guess the json is broken in some cases.
Are you using default template of catalog?

**Reply by sidiney-crescencio** [reply]
@ryota-sakamoto  Yes, the only thing I modified was the 'context' and added this in the configmap.

  service.teams: |
    recipientUrls: 
      channelName: $channel-teams-url

had to rename to .log , since .yaml is not possible, but here is the yaml config in place with my notifications deployment .

[argocd-notifications.log](https://github.com/argoproj-labs/argocd-notifications/files/7098068/argocd-notifications.log)


≈

**Reply by sidiney-crescencio** [reply]
@ryota-sakamoto  should I open an issue for this? do you think it's a bug or related to something else? Even though I'm using mostly default settings just the changes above as I have explained.

**Reply by ryota-sakamoto** [reply]
I've checked your ConfigMap, then the root cause is fixed by below issue.
https://github.com/argoproj-labs/argocd-notifications/pull/271

**Reply by sidiney-crescencio** [reply]
@ryota-sakamoto I guess I just need to wait for next release then? or is there another way to get this fix in place?

Thanks for checking it.

**Reply by ryota-sakamoto** [reply]
@sidiney-crescencio 
You can patch it in place to fix this issue yourself.
You don't need to wait next release.
